proctor-common/src/test/java/com/indeed/proctor/common/TestProctor.java

```diff
@@ -566,6 +576,206 @@ public void testDetermineTestGroups_ForceGroupsWithDefaultToMinActive() {
+                                .addAllocations(allocationX)
+                                .build());
+
+        final Map<String, ConsumableTestDefinition> tests = ImmutableMap.of("X", testDefinitionX);
+        final TestMatrixArtifact matrix = new TestMatrixArtifact();
+        matrix.setTests(tests);
+        matrix.setAudit(new Audit());
+
+        final Proctor proctor =
+                Proctor.construct(
+                        matrix,
+                        ProctorLoadResult.emptyResult(),
+                        RuleEvaluator.defaultFunctionMapperBuilder().build());
+
+        final ProctorResult proctorResult =
+                proctor.determineTestGroups(
+                        Identifiers.of(TestType.AUTHENTICATED_USER, null),
+                        ImmutableMap.of("country", "US"),
+                        ForceGroupsOptions.builder().build(),
+                        Collections.emptyList());
+
+        assertThat(proctorResult.getAllocations()).isEmpty();
+        assertThat(proctorResult.getBuckets()).isEmpty();
+        assertThat(proctorResult.getTestDefinitions()).containsEntry("X", testDefinitionX);
+    }
+
+    @Test
+    public void testDetermineTestGroups_UnitlessAllocationMissingFlag() {
+        final TestBucket inactiveBucket = new TestBucket("inactive", -1, "");
+        final TestBucket controlBucket = new TestBucket("control", 0, "");
+        final TestBucket activeBucket = new TestBucket("active", 1, "");
+        final Allocation allocationX =
+                new Allocation(
+                        "missingExperimentalUnit && country == 'US'",
+                        ImmutableList.of(new Range(0, 0), new Range(-1, 0), new Range(1, 1)));
+
+        final ConsumableTestDefinition testDefinitionX =
+                ConsumableTestDefinition.fromTestDefinition(
+                        TestDefinition.builder()
+                                .setEnableUnitlessAllocations(false)
+                                .setSalt("&X")
+                                .setTestType(TestType.AUTHENTICATED_USER)
+                                .addBuckets(inactiveBucket, controlBucket, activeBucket)
+                                .addAllocations(allocationX)
+                                .build());
+
+        final Map<String, ConsumableTestDefinition> tests = ImmutableMap.of("X", testDefinitionX);
+        final TestMatrixArtifact matrix = new TestMatrixArtifact();
+        matrix.setTests(tests);
+        matrix.setAudit(new Audit());
+
+        final Proctor proctor =
+                Proctor.construct(
+                        matrix,
+                        ProctorLoadResult.emptyResult(),
+                        RuleEvaluator.defaultFunctionMapperBuilder().build());
+
+        final ProctorResult proctorResult =
+                proctor.determineTestGroups(
+                        Identifiers.of(TestType.AUTHENTICATED_USER, null),
```

Comment:
I forgot the implementation details so I may be wrong, but isn't there a different behavior when the identifier is null (as opposed to "" or 0 etc.). It seems to be excluded here : https://github.com/indeedeng/proctor/pull/130/files#diff-620002b1ff3f4550171c6fa40c9c53bf0d567f14ba8693c5efbb185b67228c3aR314. 

If you are confident this is dealt with in the same way no concern though.

Reply:
Proctor open source only checks for null. This should be handled internally correctly. Will put up MR internally with more tests to confirm.

Reply:
Got it, thanks for the explanation!